invenio_rdm_records/records/systemfields/permission_flags.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it
+# and/or modify it under the terms of the MIT License; see LICENSE file for
+# more details.
+
+"""Permission flags system field."""
+
+from invenio_records.dictutils import dict_lookup
+from invenio_records.systemfields import SystemField
+
+
+class ParentPermissionFlagsField(SystemField):
+    """System field for managing record permissions."""
+
+    def __get__(self, record_parent, owner=None):
+        """Get the record parent's permission_flag object."""
+        if record_parent is None:
+            # returns the field itself.
+            return self
+
+        try:
+            return dict_lookup(record_parent, "permission_flags")
+        except KeyError:
+            return None
+
+    def __set__(self, record_parent, value):
+        """Set the record parent's permission_flag object."""
+        record_parent["permission_flags"] = value
```

Comment:
minor/question: could we get away with a `invenio_records.systemfields.DictField`?